Deprecate constructors with port parameter in UndertowEmbeddedServletContainer

dreis2211 opened the pull request:
Hey,

since commit e927f52f2674fe0dd85ad2a051b1bb9cacaaf975 respectively #2584 the port field in the UndertowEmbeddedServletContainer was removed, but the parameter stayed although it isn't used anymore. This PR removes the unused parameter.

I thought about deprecating the constructor first, but decided against it as the container should be instantiated via the factory and not in a standalone-fashion anyways. But if you want I could do it like that. Just let me know.

- [X] I have signed the CLA

Cheers,
Christoph

philwebb commented:
Thanks for the pull-request!

I agree that the container _should_ be instantiated via the factory, but since the old constructors are public we should probably do at least one release with them deprecated rather than removed. Would you be able to rework the PR?

dreis2211 commented:
As mentioned I thought about it already, so no problem at all.

Added the constructors again. Also added some Javadocs in order to properly document that the new constructors without port argument should be used.

wilkinsona commented:
Merged. Thanks again for the pull request.
